geode-redis/src/distributedTest/java/org/apache/geode/redis/executors/ExpireDUnitTest.java

```diff
@@ -0,0 +1,233 @@
+    serverProperties3.setProperty(REDIS_BIND_ADDRESS, LOCAL_HOST);
+
+    locator = clusterStartUp.startLocatorVM(0, locatorProperties);
+    server1 = clusterStartUp.startServerVM(1, serverProperties1, locator.getPort());
+    server2 = clusterStartUp.startServerVM(2, serverProperties2, locator.getPort());
+    server3 = clusterStartUp.startServerVM(3, serverProperties3, locator.getPort());
+
+    jedis1 = new Jedis(LOCAL_HOST, availablePorts[0], JEDIS_TIMEOUT);
+    jedis2 = new Jedis(LOCAL_HOST, availablePorts[1], JEDIS_TIMEOUT);
+    jedis3 = new Jedis(LOCAL_HOST, availablePorts[2], JEDIS_TIMEOUT);
+  }
+
+  @After
+  public void testCleanUp() {
+    jedis1.flushAll();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    jedis1.disconnect();
+    jedis2.disconnect();
+    jedis3.disconnect();
+
+    server1.stop();
+    server2.stop();
+    server3.stop();
+  }
+
+  @Test
+  public void expireOnOneServer_shouldPropagateToAllServers() {
+    String key = "key";
+
+    jedis1.set(key, "value");
+    jedis1.expire(key, 20);
+
+    assertThat(jedis2.ttl(key)).isGreaterThan(0);
+  }
+
+  @Test
+  public void expireOnOneServer_shouldResultInKeyRemovalFromOtherServer() {
+    String key = "key";
+
+    jedis1.set(key, "value");
+    jedis1.expire(key, 2);
+
+    assertThat(jedis2.ttl(key)).isGreaterThan(0);
+    GeodeAwaitility.await().until(() -> jedis2.ttl(key) == TTLExecutor.NOT_EXISTS);
+  }
+
+  @Test
+  public void whenExpirationIsSet_andIsUpdatedOnAnotherServer_itIsReflectedOnFirstServer() {
+    String key = "key";
+
+    jedis1.set(key, "value");
+    jedis1.expire(key, 20);
+    jedis2.expire(key, 10000);
+
+    assertThat(jedis1.ttl(key)).isGreaterThan(20L);
+  }
+
+  @Test
+  public void whenExpirationIsSet_andKeyIsResetOnAnotherServer_ttlIsRemovedOnFirstServer() {
+    String key = "key";
+
+    jedis1.set(key, "value");
+    jedis1.expire(key, 20);
+    jedis2.set(key, "newValue");
+
+    assertThat(jedis1.ttl(key)).isEqualTo(TTLExecutor.NO_TIMEOUT);
+  }
+
+  @Test
+  public void whenExpirationIsSet_andKeyIsPersistedOnAnotherServer_ttlIsRemovedOnFirstServer() {
+    String key = "key";
+
+    jedis1.set(key, "value");
+    jedis1.expire(key, 20);
+    jedis2.persist(key);
+
+    assertThat(jedis1.ttl(key)).isEqualTo(TTLExecutor.NO_TIMEOUT);
+  }
+
+  @Test
+  public void whenExpirationIsSet_andKeyIsDeletedOnAnotherServer_ttlReflectsChanges() {
+    String key = "key";
+
+    jedis1.set(key, "value");
+    jedis1.expire(key, 20);
+    jedis2.del(key);
+
+    server1.invoke(() -> {
+      InternalCache cache = ClusterStartupRule.getCache();
+      GeodeRedisService redisService = cache.getService(GeodeRedisService.class);
+      boolean hasExpiration = redisService.getGeodeRedisServer().getRegionCache()
+          .hasExpiration(new ByteArrayWrapper(key.getBytes()));
+      assertThat(hasExpiration).as("expiration should not be set").isFalse();
```

Comment:
why do we need to check the internal state in the integration test?  everything else operates at the external, `jedis` level.

Reply:
Good point.  The TTL was returning non-existent for server1, but internally the region still has an expiration set.  Updated it so we reset the key on server2 and check the TTL without looking into the region cache.